clap_complete/tests/testsuite/fish.rs

```diff
@@ -119,3 +119,29 @@ fn subcommand_last() {
         name,
     );
 }
+
+#[test]
+#[cfg(unix)]
+fn register_completion() {
+    if !common::has_command("fish") {
+        return;
+    }
+
+    common::register_example("test", completest::Shell::Fish);
+}
+
+#[test]
+#[cfg(unix)]
+fn complete() {
+    if !common::has_command("fish") {
+        return;
+    }
+
+    let term = completest::Term::new();
+    let runtime = common::load_runtime("test", completest::Shell::Fish);
+
+    let input = "test \t";
+    let expected = r#"% test "#;
```

Comment:
Unsure if the vt100 parser I'm using is being too good at its job but I'm not getting any output.  I have put in debug code to intercept the messages being read from the shell, before we parse, and the output exists.

```rust
"\u{1b}[?2004h"
"\u{1b}[30m\u{1b}[m\u{f}"
"\r"
"% \u{1b}[K\r\u{1b}[C\u{1b}[C"
"test \r\u{1b}[7C"
"\r\naction  help  (\u{1b}[33mPrint this message or the help of the given subcommand(s)\u{1b}[30m\u{1b}[m\u{f})  last    quote\r\nalias   hint
               pacman  value\u{1b}M\u{1b}M\r\u{1b}[7C"
"\u{8}\u{8}\u{8}\u{8}\u{8}\u{1b}[38;2;0;95;215mtest\u{1b}[30m\u{1b}[m\u{f} \r\u{1b}[7C"
"\r\n\u{1b}[J\u{1b}[30m\u{1b}[m\u{f}\u{1b}[30m\u{1b}[m\u{f}"
"\u{1b}[?2004l"
```

Reply:
I switched to bpaf's parser.  It looks like the difference is its ignoring of `clear_screen`

However, there appears to be a race in elvish now with showing `COMPLETING argument`

Reply:
As far as I remember it was something to do with the relative screen scroll and it ignores not clear the screen in general but usually some fraction of it. At that point I had something that worked and I had no goal of making a finished product of it just yet (pending rewrite, etc) and the goal was to use it in making documentation. At some point I wanted to take a look at sources of fish to see what exactly is it trying to do here.

Race... Problem with talking to a program over pte is that you have no idea when it's done so comptester spawns a separate thread with the reader and uses "if it said something then been quiet over time T then it's probably done". If your (or CI) machines are slower - increasing that T might help. This is why I was talking about controllable amount of concurrency - it is fine to spawn more tests at once and having them finish within T since most of them are going to be sleeping.

Try bumping T?

Reply:
I think the race is something different.   My suspicion is that when elvish detects the completions are taking too long, it prints a message to the user telling them what is happening.  When using a full vte parser, this message is cleared.  When we ignore screen clearing, it will show up depending on available compute

Reply:
Interesting. Haven't seen this one. I'll look into adding screen cleaning back once I'm done with the documentation update.

Reply:
My guess is that the dynamic completions of bpaf are slower than the pre-computer compeletions of `clap_complete` so you always get the message.

Reply:
What message are we talking about? I don't recall seeing anything that is not supposed to be there as far as completions are concerned at least.

Reply:
"COMPLETING argument"... Yea, have this one.

Reply:
https://elv.sh/ref/edit.html - it's in their documentation though and in the examples all over the internet. I think that's intended behavior...